src/types.ts

```diff
@@ -234,6 +235,26 @@ function string<Options extends StringOptions>(options?: Options): GetType<strin
   } as unknown as GetType<string, Options>;
 }
 
+function tuple<Items extends readonly unknown[], Options extends GenericOptions>(
+  items: Items,
+  options?: Options
+): GetType<TupleItems<Items>, Options> {
+  const { required } = options || {};
+
+  const minItems = items.reduce<number>((value, item: { $required: boolean }, index) => {
+    if (item.$required) return index + 1;
+    return value;
+  }, 0);
+
+  return {
+    ...(required ? { $required: true } : {}),
+    items,
```

Comment:
The Draft 4 mentioned [here](https://json-schema.org/understanding-json-schema/reference/array.html#tuple-validation) seems to support `prefixItems` to support these tuple structure. Do you know if we can use this instead and if the MongoDB server supports this?

Reply:
> In Draft 4 - 2019-09, tuple validation was handled by an alternate form of the `items` keyword. When `items` was an array of schemas instead of a single schema, it behaved the way `prefixItems` behaves.

My understanding of that is in Draft 4 `prefixItems` isn't supported, and instead tuple validation was achieved with the `items` keyword. 

Looking at the [Draft 4 validation spec](https://datatracker.ietf.org/doc/html/draft-fge-json-schema-validation-00#section-5.3.1) confirms this:

> [5.3.1](https://datatracker.ietf.org/doc/html/draft-fge-json-schema-validation-00#section-5.3.1).  additionalItems and items
>
>[5.3.1.1](https://datatracker.ietf.org/doc/html/draft-fge-json-schema-validation-00#section-5.3.1.1).  Valid values
>
>   The value of "additionalItems" MUST be either a boolean or an object.
>   If it is an object, this object MUST be a valid JSON Schema.
>
>   The value of "items" MUST be either an object or an array.  If it is
>   an object, this object MUST be a valid JSON Schema.  If it is an
>   array, items of this array MUST be objects, and each of these objects
>   MUST be a valid JSON Schema.

If we were to introduce the ability to add additional properties beyond the defined tuple properties (we're not currently doing this) we would need to use the `additionalItems` keyword.